WordPress/gutenberg
# Block Locking: Register the 'lock' attribute on the server

## What?
Resolves #40461.

PR fixes an error when locking server-side rendered blocks.

## Why?
The `block-renderer` endpoint validates passed attributes, but the `lock` block attribute was registered on the client-side.

## How?

Use `register_block_type_args` filter to register the `lock` attribute.

## Testing Instructions
 1. Open a Post or Page.
2. Insert the Latest Comments or any block that uses the `ServerSideRender` component.
3. Lock the block using block Options.
4. Confirm there's no "This block has encountered an error and cannot be previewed" error.

## Screenshots or screencast 
| Before | After |
| --- | --- |
|![CleanShot 2022-04-20 at 15 15 09](https://user-images.githubusercontent.com/240569/164220232-8a6919e3-d5b9-488b-8f48-a5a8d076a9c8.png)|![CleanShot 2022-04-20 at 15 15 25](https://user-images.githubusercontent.com/240569/164220238-f37aef96-2772-4aa2-aaaa-63b4ac41e143.png)|
